src/Language/PureScript/Sugar/Names/Env.hs

```diff
@@ -229,13 +229,16 @@ mkPrimExports ts cs =
     , exportedTypeClasses = M.fromList $ mkClassEntry `map` M.keys cs
     }
   where
-  mkTypeEntry (Qualified mn name) = (name, ([], primExportSource mn))
-  mkClassEntry (Qualified mn name) = (name, primExportSource mn)
+  mkTypeEntry (Qualified (ByModuleName mn) name) = (name, ([], primExportSource mn))
+  mkTypeEntry _ = internalError "mkPrimExports.mkTypEntry: invalid argument"
+
+  mkClassEntry (Qualified (ByModuleName mn) name) = (name, primExportSource mn)
+  mkClassEntry _ = internalError "mkPrimExports.mkClassEntry: invalid argument"
```

Comment:
> Or does that affect a lot of error messages in our golden tests?
These are internal compiler errors so not really. Although yep, this could be helped by updating the error messages.